components/src/core/components/Form/Form.vue

```diff
@@ -38,43 +26,42 @@ export default defineComponent({
     },
   },
 
-  components: {
-    'oxd-loading-spinner': Spinner,
-  },
+  setup(props, context) {
+    const {validate, errorbag, fieldset} = useFormValidation();
 
-  computed: {
-    isValid(): boolean {
-      return this.errorBag.length === 0;
-    },
-  },
+    const isProcessing = computed(() => {
+      return fieldset.value.reduce((acc, field) => {
+        return acc && field.processing.value;
+      }, true);
+    });
 
-  methods: {
-    validate(e: Event) {
-      if (!this.loading) {
-        setTimeout(() => {
-          if (this.isValid) {
-            this.$emit('submitValid', e);
-          }
-        }, 0);
-      }
-    },
-    searchErrors(id: string) {
-      return this.errorBag.findIndex((item: ErrorField) => {
-        return item.cid == id;
-      });
-    },
-    addError(errorField: ErrorField) {
-      const i: number = this.searchErrors(errorField.cid);
-      if (i < 0) {
-        this.errorBag.push(errorField);
-      }
-    },
-    removeError(errorField: ErrorField) {
-      const i: number = this.searchErrors(errorField.cid);
-      if (i > -1) {
-        this.errorBag.splice(i, 1);
+    const isFromInvalid = computed(() => {
+      return errorbag.value.length > 0;
+    });
+
+    const isFormBusy = computed(() => {
+      return props.loading && isProcessing;
+    });
+
+    const onSubmit = async (e: Event) => {
+      if (!isFormBusy) {
+        await validate();
+        if (!isFromInvalid) {
+          context.emit('submitValid', e);
```

Comment:
@samanthajayasinghe @RajithaKumara There is a issue with parent component not receiving `submitValid` event. this is suspected to be caused due to `async validate()`. Researching more on the matter

Reply:
Identified and fixed this issue. there were 2 reasons why the events were not received by parent component. 1 - There was a mistake in code where I expected form's `validate()` fn to resolve a promise and carry the execution. but by the time it execute, it's not resolving but returning an already resolved promise. 2 - Removed async keyword from `onSubmit()` fn and made it a regular js function by wrapping it with `timeout` fn with offset 0. this will guarantee the execution of code without creating any race condition. ( [why setTimeout(fn, 0) sometimes useful?](https://stackoverflow.com/questions/779379/why-is-settimeoutfn-0-sometimes-useful) )